mdtraj/core/topology.py

```diff
@@ -592,6 +593,9 @@ def from_dataframe(
             if col not in atoms.columns:
                 raise ValueError("dataframe must have column %s" % col)
 
+        if "formal_charge" not in atoms.columns:
```

Comment:
I'm not a huge fan of how this function modifies the user's dataframe, but there is already a precedent for this with `segmentID`

Reply:
It was modifying the dataframe before, so I think it's okay to *not* make a copy. It would change the behavior of the function, so let's leave it for now.